spec/requests/users_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Users API', type: :request do
+  let(:valid_user_params) do
+    {
+      user: {
+        name: Faker::Name.name,
+        email: Faker::Internet.email,
+        password: 'Password123+',
+        password_confirmation: 'Password123+'
+      }
+    }
+  end
+
+  describe 'Signup' do
+    let(:headers) { { 'ACCEPT' => 'application/json' } }
+
+    context 'valid user params' do
+      it 'creates a user account for valid params' do
+        expect { post API_ENDPOINTS[:Users][:Signup], params: valid_user_params, headers: }.to change(User, :count).from(0).to(1)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response).to have_http_status(:created)
+        expect(JSON.parse(response.body)['errors']).to be_empty
+      end
+    end
+
+    context 'invalid user params' do
+      it 'fails for missing params' do
+        invalid_user_params = { user: {} }
+        expect { post API_ENDPOINTS[:Users][:Signup], params: invalid_user_params, headers: }.not_to change(User, :count)
+        expect(response.content_type).to eq('application/json; charset=utf-8')
+        expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors']).not_to be_empty
```

Comment:
I would actually check that a specific error is returned. Maybe omit 1 param and check the `errors` array

Reply:
You got reason.
What about deep matching the user erro4s array with the errors array recieved in the response body?
I think it solves this concern?

Reply:
I'd still explicitly check the error is what we expect it. What if the error returned here is "password confirmation doesn't match" or "email is already taken" or any other error?

Our test is specifically for missing params so we should explicitly check for that. Otherwise, this is just a test that check that those params return an error (doesn't matter which one)

Reply:
But after giving it a second thought it feels that the word "missing" isn't logical because the checks are done by the model preset validations and has nothing to do with some parameters missing or not.

I think I should change test description to invalid parameters instead of missing because that's what it does.
And I'll make sure the errors in the response matches those of the model validation with the same values.
What do you think?

Reply:
Yeah that's another option. We simply test that invalid params = fail and not get into the specifics of why or how. 

I think that makes more sense as well

Reply:
Perfect, I'll change it either.
Thanks for your review!

Reply:
Done